packages/flutter_tools/lib/src/web/bootstrap.dart

```diff
@@ -216,11 +231,82 @@ let _scriptUrls = {
     window.\$dartLoader.loadConfig = loadConfig;
     window.\$dartLoader.loader = loader;
 
-    // TODO(srujzs): Support hot restart.
-
     // Begin loading libraries
     loader.nextAttempt();
-  }
+
+    // Set up stack trace mapper.
+    if (window.\$dartStackTraceUtility &&
+        !window.\$dartStackTraceUtility.ready) {
+      window.\$dartStackTraceUtility.ready = true;
+      window.\$dartStackTraceUtility.setSourceMapProvider(function(url) {
+        var baseUrl = window.location.protocol + '//' + window.location.host;
+        url = url.replace(baseUrl + '/', '');
+        if (url == 'dart_sdk.js') {
+          return dartDevEmbedder.debugger.getSourceMap('dart_sdk');
+        }
+        url = url.replace(".lib.js", "");
```

Comment:
Where is the coupling here? The ".lib.js" in the file name is configured somewhere in the compile actions?

Reply:
Alas, the frontend server hard-codes it in: https://github.com/dart-lang/sdk/blob/ba9ef09163b97992dacc3a18b5f8846958818cf8/pkg/frontend_server/lib/src/javascript_bundle.dart#L270, so there's an implicit dependency there. I don't think there's a way in the recompile to tell it what to use.